message/producer.go

```diff
@@ -2,29 +2,39 @@ package message
 
 import (
 	"github.com/ONSdigital/dp-dimension-importer/event"
-	"github.com/ONSdigital/go-ns/kafka"
 	"github.com/ONSdigital/go-ns/log"
 )
 
 const (
-	marshalErr = "Unexpected error while attempting to avro marshall event.DimensionsInsertedEvent"
+	marshalErr = "Unexpected error while attempting to avro marshall event.InstanceCompletedEvent"
 )
 
-//go:generate moq -out ./message_test/producer_generated_mocks.go -pkg message_test . Marshaller
+//go:generate moq -out ./message_test/producer_generated_mocks.go -pkg message_test . Marshaller KafkaProducer
+
+type KafkaProducer interface {
+	Output() chan []byte
+}
 
 // Marshaller defines a type for marshalling the requested object into the required format.
 type Marshaller interface {
 	Marshal(s interface{}) ([]byte, error)
 }
 
-// DimensionInsertedProducer type for producing DimensionsInsertedEvents to a kafka topic.
-type DimensionInsertedProducer struct {
+// InstanceCompletedProducer type for producing DimensionsInsertedEvents to a kafka topic.
+type InstanceCompletedProducer struct {
 	Marshaller Marshaller
-	Producer   kafka.MessageProducer
+	Producer   KafkaProducer
+}
+
+func NewInstanceCompletedProducer(producer KafkaProducer, marshaller Marshaller) InstanceCompletedProducer {
+	return InstanceCompletedProducer{
+		Producer:   producer,
+		Marshaller: marshaller,
+	}
 }
 
-// DimensionInserted kafka message to complete dimension inserted event
-func (p DimensionInsertedProducer) DimensionInserted(e event.DimensionsInsertedEvent) error {
+// Completed kafka message to complete dimension inserted event
```

Comment:
Unclear whether this comment should still reference dimension inserted event or should talk about instance completed instead?